<launch>

  <arg name="prefix" />
  <arg name="rviz" default="true" />

  <node name="look_around_bin_main"
        pkg="label_octomap" type="look-around-bin.l"
        required="true">
  </node>
  <node name="rosbag_record"
        pkg="rosbag" type="record"
        args="
              -b 0
              -O $(optenv HOME)/.ros/label_octomap/$(arg prefix)_look_around_bin.bag

              /tf
              /robot/joint_states
              /look_around_bin_main/output/start
              /look_around_bin_main/output/stop

              /right_hand_camera/rgb/camera_info
              /right_hand_camera/rgb/image_raw
              /right_hand_camera/depth_registered/camera_info
              /right_hand_camera/depth_registered/image_raw
              ">
  </node>

  <group if="$(arg rviz)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find label_octomap)/rvizconfig/look_around_bin.rviz">
    </node>
  </group>


</launch>
